neuro_san/internals/messages/message_utils.py

```diff
@@ -1,277 +0,0 @@
-
-# Copyright (C) 2023-2025 Cognizant Digital Business, Evolutionary AI.
-# All Rights Reserved.
-# Issued under the Academic Public License.
-#
-# You can be released from the terms, and requirements of the Academic Public
-# License by purchasing a commercial license.
-# Purchase of a commercial license is mandatory for any use of the
-# neuro-san SDK Software in commercial settings.
-#
-# END COPYRIGHT
-
-from typing import Any
-from typing import Dict
-from typing import List
-from typing import Tuple
-from typing import Union
-
-import json
-
-from langchain_core.messages.ai import AIMessage
-from langchain_core.messages.base import BaseMessage
-from langchain_core.messages.human import HumanMessage
-from langchain_core.messages.system import SystemMessage
-from langchain_core.messages.tool import ToolMessage
-
-from neuro_san.internals.messages.agent_message import AgentMessage
-from neuro_san.internals.messages.agent_framework_message import AgentFrameworkMessage
-from neuro_san.internals.messages.agent_tool_result_message import AgentToolResultMessage
-from neuro_san.internals.messages.chat_message_type import ChatMessageType
-
-
-def pretty_the_messages(messages: List[Any]) -> str:
-    """
-    Pretty printing helper
-    :param messages: A list of OpenAI messages
-    :return: A single string representing the output of the messages.
-    """
-    prettied = ""
-    messages_list = list(messages)  # Convert to list to allow reversing
-
-    # Iterate in reverse to find the last user message
-    rm = []
-    for m in reversed(messages_list):
-        rm.append(m)
-        if get_role(m) == "user":
-            break
-
-    for m in reversed(rm):
-        if any(get_content(m)):
-            prettied += f"{get_role(m)}: {get_content(m)}\n"
-
-    return prettied
-
-
-def get_last_message_with_content(messages: List[Any]) -> object:
-    """
-    Sometimes just indexing a message list by [-1]
-    gives you a message that does not actually have any content.
-    This method gives you the last messages that does have content.
-
-    :param messages: input list of OpenAI messages
-    :return: An applicalble OpenAI message (or None if no message is applicable)
-    """
-
-    messages_list = list(messages)
-
-    for m in reversed(messages_list):
-        if any(get_content(m)):
-            return m
-
-    return None
-
-
-def generate_response(the_messages: List[Any]) -> str:
```

Comment:
This method moves to IntraAgentMessageUtils.
It was also refactored a bit in there for clarity and fewer repeated calls to the same methods.